### ISSUE TITLE
Handle HTTP 429 for Fedi APIs

### ISSUE THREAD

**Issue by robertoszek** [robertoszek/pleroma-bot#75]
Handle rate limits if HTTP 429 Too Many Requests is raised when sending requests to the Pleroma/Mastodon/Misskey API endpoints.

**Comment by robertoszek** [comment]
Mastodon seems pretty straightforward:
https://docs.joinmastodon.org/api/rate-limits/

However, we may have to fallback to sleeping for an arbitrary amount of time for Pleroma and Misskey.

In my testing they don't seem to return usable headers when an HTTP 429 is raised (maybe due to Pleroma instances being usually served with a reverse proxy). I may also try to test against a Misskey instance which is not behind Cloudflare.

But either way, if no usable headers are found, a fallback is needed.

**Comment by nemobis** [comment]
Example:
```
Traceback (most recent call last):
  File "/home/7/federico/mastodon/lib/python3.7/site-packages/pleroma_bot/_pleroma.py", line 125, in post_pleroma
    response.raise_for_status()
  File "/home/7/federico/mastodon/lib/python3.7/site-packages/requests/models.py", line 960, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 429 Client Error: Too Many Requests for url: https://respublicae.eu/api/v1/media

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/7/federico/mastodon/lib/python3.7/site-packages/pleroma_bot/cli.py", line 554, in main                                                                                                                                       
    tweets_to_post["media_processed"]
  File "/home/7/federico/mastodon/lib/python3.7/site-packages/pleroma_bot/_utils.py", line 620, in post
    post_id = self.post_pleroma(tweet, poll, sensitive, media)
  File "/home/7/federico/mastodon/lib/python3.7/site-packages/pleroma_bot/_pleroma.py", line 152, in post_pleroma
    response.raise_for_status()
  File "/home/7/federico/mastodon/lib/python3.7/site-packages/requests/models.py", line 960, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 429 Client Error: Too Many Requests for url: https://respublicae.eu/api/v1/media
```

**Comment by robertoszek** [comment]
I've committed these changes to the develop branch:
https://github.com/robertoszek/pleroma-bot/commit/2d90b9e6488cdda7bb0fca7da4bb1cbde2c3f164

They are included in `1.1.1rc31`:
`pip install -i https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple pleroma-bot==1.1.1rc31`

Unfortunately I didn't find any usable headers for Misskey, I may add a retry limit and sleep 10 secs between requests if we hit an 429 for Misskey instances in the future.
